util/src/test-support/java/tech/pegasys/artemis/util/executors/MockScheduledExecutor.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.util.executors;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.Delayed;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class MockScheduledExecutor extends MockExecutorService implements ScheduledExecutorService {
```

Comment:
We have an `AsyncRunner` interface now, (along with `DelayedExecutorAsyncRunner` and `StubAsyncRunner`) now so `MockScheduledExecutor` might be overkill for now, or else, we can remove the `AsyncRunner`, and refactor to use `MockScheduledExecutor`. In any case, we should pick one to prevent duplication.

With AsyncRunner interface, this is how the `connect()` method in PeerManager.java would look like:
```
  public SafeFuture<?> connect(final Multiaddr peer, final Network network) {
    STDOUT.log(Level.DEBUG, "Connecting to " + peer);
    return SafeFuture.of(network.connect(peer))
        .whenComplete(
            (conn, throwable) -> {
              if (throwable != null) {
                STDOUT.log(
                    Level.DEBUG,
                    "Connection to " + peer + " failed. Will retry shortly: " + throwable);
                    asyncRunner.runAfterDelay(
                        () -> connect(peer, network),
                        RECONNECT_TIMEOUT,
                        TimeUnit.MILLISECONDS);
              } else {
                STDOUT.log(
                    Level.DEBUG,
                    "Connection to peer: "
                        + conn.secureSession().getRemoteId()
                        + " was successful");
                SafeFuture.of(conn.closeFuture())
                    .finish(
                        () -> {
                          LOG.debug("Connection to {} closed. Will retry shortly", peer);
                              asyncRunner.runAfterDelay(
                                  () -> connect(peer, network),
                                  RECONNECT_TIMEOUT,
                                  TimeUnit.MILLISECONDS));
                        });
              }
            });
  }
```

Reply:
Makes sense - will look into this